router_api.go

```diff
@@ -4,9 +4,26 @@ import (
 	"encoding/json"
 	"net/http"
 	"runtime"
+	"time"
 )
 
-func newAPIHandler(rout *Router) http.Handler {
+func newAPIHandler(rout *Router) (api http.Handler, err error) {
+	reloadDuration, err := time.ParseDuration(reloadInterval)
+	if err != nil {
+		return nil, err
+	}
+	reloadChan := make(chan bool)
+	go func() {
+		// Rate-limit reloads to 1 per RELOAD_INTERVAL.
+		// This goroutine blocks until it receives a message on reloadChan, then
+		// waits for the timeout before calling reload.
+		for {
+			<-reloadChan
```

Comment:
This would be more idiomatic as:

```go
for range reloadChan {
    //...
}
```
See range clauses in the [for statement documentation](http://golang.org/ref/spec#For_statements) for more details.